app/controllers/agenda/agendaitems.js

```diff
@@ -97,7 +98,25 @@ export default class AgendaAgendaitemsController extends Controller {
   }
 
   @task
-  *assignNewPriorities(reorderedAgendaitems) {
+  *assignNewPriorities(reorderedAgendaitems, agendaitemType) {
+    const agendaitemsCount = yield this.store.count('agendaitem', {
+      'filter[agenda][:id:]': this.agenda.id,
+      'filter[type][:uri:]': agendaitemType,
+    });
+    // concurrency check if agendaitems were added or removed
+    if (reorderedAgendaitems?.length !== agendaitemsCount) {
+      this.toaster.error(
+        this.intl.t('agendaitem-positions-cannot-be-changed'),
+        this.intl.t('changes-could-not-be-saved-title'),
+        {
+          timeOut: 60000,
+        }
+      );
+      // we don't want the changes anymore, but we are not going to refresh the page
+      this.notasHasChanged = false;
+      this.announcementsHasChanged = false;
+      return;
+    }
```

Comment:
high risk scenario:
User 1 and 2 have the same agenda loaded.
User 1 accepts a submission (f.e. should become agendaitem number 2)
User 2 changes the agendaitem positions without refreshing (swap "stale" agendaitem 2 and 3 (in same mandatee block)
There will now be 2 agendaitems with number 2. and number 4 is missing.
Similar cases in case of deletes/send back to submitter
This prevents such issues, not allowing based the amount of items from ask query vs what we have in local stores.
Not foolproof (1 addition and 1 deletion will allow editing) but deletions are way less frequent than additions.